### PR TITLE
[node] add EventEmitter.setMaxListeners static method

### PR DESCRIPTION
**Pull request by ejose19**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/pull/36001

### CODE REVIEW

File: types/node/events.d.ts [hunk 1]
```diff
@@ -260,6 +268,26 @@ declare module 'events' {
...
+         * EventEmitter.defaultMaxListeners = 20;
+         *
+         * const eventTarget = new EventTarget();
+         * // Only way to increase limit for `EventTarget` instances
+         * // as these doesn't expose its own `setMaxListeners` method
+         * EventEmitter.setMaxListeners(20, eventTarget);
+         * ```
+         * @since v15.3.0, v14.17.0
+         */
+        static setMaxListeners(n?: number, ...eventTargets: DOMEventTarget[]): void;
```
Comment (peterblazejewicz) [inline types/node/events.d.ts:290]: Shouldn't this allow both types, as in documentation and ,say, in `getEventListeners`?:
```ts
static getEventListeners(emitter: DOMEventTarget | NodeJS.EventEmitter
```
we support v14, v16 and current, so this should added all versions (as per support coment)

File: types/node/events.d.ts [hunk 2]
```diff
@@ -154,7 +154,11 @@ declare module 'events' {
          * ```
          * @since v11.13.0, v10.16.0
          */
-        static once(emitter: NodeEventTarget, eventName: string | symbol, options?: StaticEventEmitterOptions): Promise<any[]>;
+        static once(
```
Comment (SimonSchick) [inline types/node/events.d.ts:157]: Please refrain from making unrelated changes.

File: types/node/events.d.ts (hunk 1 above)
Comment (SimonSchick) [reply types/node/events.d.ts:290]: Yes it should support EventEmitters too.

Comment (SimonSchick) [reply types/node/events.d.ts:290]: Also please ensure to add the note what happens when `eventTargets` is empty.

**Review by SimonSchick — CHANGES_REQUESTED** [review]

File: types/node/events.d.ts (hunk 2 above)
Comment (ejose19) [reply types/node/events.d.ts:157]: Actually, that change is made by prettier, if these are not fixed, every contributor will have to manually revert every style fixes which adds a burden on contributing, I can revert if that's what you end deciding.

**Comment by sandersn** [comment]
@SimonSchick do the recent changes address your comments?

**Review by gabritto — APPROVED** [review]
